lunatrace/bsl/frontend/src/utils/users.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright by LunaSec (owned by Refinery Labs, Inc)
+ *
+ * Licensed under the Business Source License v1.1
+ * (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ * https://github.com/lunasec-io/lunasec/blob/master/licenses/BSL-LunaTrace.txt
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import { GetSidebarInfoQuery } from '../api/generated';
+import { impersonateUserKey } from '../constants/localstorage';
+import { ImpersonateUser } from '../types/user';
+
+export function getImpersonatedUser(): ImpersonateUser | null {
+  const user = localStorage.getItem(impersonateUserKey);
```

Comment:
This is ok but we actually already have the code to do this in the app, and its a bit better because it uses a hook to do it:
https://github.com/lunasec-io/lunasec/blob/6fc6fa0e08234f0032ab570b496eec4c2830e53d/lunatrace/bsl/frontend/src/hooks/useLocalStorage.ts#L16

You can see this strategy being used from some of our contexts/providers as well.  If you only use it in one place, then you can just call that hook directly from the component that needs it.  If you want to use it in multiple places and have it rereder automatically on change, then the way to do that is with a provider. In that case, you would write a context called `ImpersonatedUser` that provides this data and is responsible for mapping it into localStorage using the `useLocalStorage` hook.  Then, any changes will cascade through the app/DOM automatically, without a component having to know to reload this information out of localStorage.  Thats only if you want to share this state app-wide. You can see an example with this context provider https://github.com/lunasec-io/lunasec/blob/master/lunatrace/bsl/frontend/src/contexts/LayoutContext.tsx

So yeah, TLDR, try to use the machinery we have and do this the 'reacty' way.